scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,230 @@
+import numpy as np
+from collections import namedtuple
+from ._axis_nan_policy import _axis_nan_policy_factory
+
+
+def calculate_bws_sum(ranks, nm):
+    '''Calculate the sum for the Baumgartner-Weiss-Schindler statistic'''
+    sum = 0
+    size = len(ranks)
+    factor = 1 / (size + 1)
+    for i, r in enumerate(ranks):
+        # index & ranks are 0-based, but we want 1-based
+        i1, r1 = i + 1, r + 1
+        sum += (r1 - nm / size * i1) ** 2 / (i1 * factor * (1 - i1 * factor))
+    return sum
```

Comment:
This is going to be very slow. Use numpy instead. 

```suggestion
def calculate_bws_sum(ranks, nm):
    '''Calculate the sum for the Baumgartner-Weiss-Schindler statistic'''
    r1 = ranks + 1
    size = len(ranks)
    factor = 1 / (size + 1)
    i1 = np.arange(1, size + 1)
    ifact = i1 * factor
    return ((r1 - (nm/size)*i1) ** 2 / (ifact*(1-ifact))).sum()
```